services/src/main/java/org/broadleafcommerce/payment/service/gateway/DefaultPayPalCheckoutExternalCallService.java

```diff
@@ -258,17 +272,29 @@ public ItemList getPayPalItemList(PaymentRequest paymentRequest,
     public Amount getPayPalAmountFromOrder(PaymentRequest paymentRequest) {
         Details details = new Details();
 
-        details.setShipping(Objects.toString(paymentRequest.getShippingTotal(), null));
-        details.setSubtotal(Objects.toString(paymentRequest.getOrderSubtotal(), null));
-        details.setTax(Objects.toString(paymentRequest.getTaxTotal(), null));
+        details.setShipping(getStringAmount(paymentRequest.getShippingTotal()));
+        details.setSubtotal(getStringAmount(paymentRequest.getOrderSubtotal()));
+        details.setTax(getStringAmount(paymentRequest.getTaxTotal()));
 
         Amount amount = new Amount();
         amount.setCurrency(paymentRequest.getCurrencyCode());
-        amount.setTotal(Objects.toString(paymentRequest.getTransactionTotal(), null));
+        amount.setTotal(getStringAmount(paymentRequest.getTransactionTotal()));
         amount.setDetails(details);
         return amount;
     }
 
+    private String getStringAmount(BigDecimal amount) {
+        return Objects.toString(normalizePrice(amount), null);
+    }
+
+    private BigDecimal normalizePrice(BigDecimal amount) {
+        if (amount == null) {
+            return null;
+        }
+
+        return amount.setScale(2, RoundingMode.HALF_EVEN);
```

Comment:
@charchar @nathandmoore Is this RoundingMode consistent with the everything else that we've built? 

Is there a way to globally declare the rounding mode or a single service/util that centralizes this logic?

Reply:
I just realized that this change is in `blc-paypal` which changes the context a bit, but the questions above are still relevant. Maybe CartOpServices is responsible for passing in values that are correctly rounded?

Reply:
I think you are right, but actually this is required only by the PayPal API [PayPal API](https://developer.paypal.com/docs/api/payments/v1/#definition-amount) so I'm not sure. I think this should be some kind of general solution, something like the `Money` class instead of `BigDecimal`.

Reply:
There's another issue about figuring out rounding in a neutral location that has yet to be tackled or prioritized https://github.com/BroadleafCommerce/CartOperationServices/issues/69

Reply:
Sounds good. @dmyroniuk can you make sure that this occurrence gets highlighted on that ticket?

Reply:
Done